### PR TITLE
Implementation of the `drag and drop` feature for files uploading

### PR DESCRIPTION
**Pull request by Oleh-Stefanovych**
Implementation of `drag and drop` feature.

The ESLint result: 
image

### CODE REVIEW

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, PTAL.

File: app/component/file-list/index.js [hunk 1]
```diff
@@ -38,13 +61,70 @@ export class FileList extends Component {
       this._files.forEach((file) => {
         new FileItem(this._root, {
           file: file,
+          classes: ['dropable'],
```
Comment (dkalinin) [inline app/component/file-list/index.js]: droppable

File: app/component/high-order-component/files-page/index.js [hunk 2]
```diff
@@ -45,8 +50,11 @@ export class FilesPage extends HighOrderComponent {
 
     this._fileListComponent = new FileList(this.element, {
       files: this.store.state.files,
+      classes: ['dropable'],
```
Comment (dkalinin) [inline app/component/high-order-component/files-page/index.js]: droppable

File: app/css/style.css [hunk 3]
```diff
@@ -0,0 +1,14 @@
+.dropable{
```
Comment (dkalinin) [inline app/css/style.css]: droppable

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, fixed, PTAL.

File: app/component/index.js [hunk 4]
```diff
@@ -59,6 +59,16 @@ export class Component {
     this._render();
   }
 
+  /**
+   * Sets a classes to the current {@link Element}.
```
Comment (dkalinin) [inline app/component/index.js]: Sets classes...

File: app/store/actions.js [hunk 5]
```diff
@@ -32,4 +31,14 @@ export default {
...
     };
   },
+
+  uploadFiles(store) {
+    return (files, targetFolder) => {
+      store.commit('setFileListStatus', 'Uploading new files...');
+      store.application.services.fileRequestService.uploadFile(files, targetFolder)
+          .then((response) => {
+            store.dispatch('fetchFiles', store.state.currentFolder);
+          });
```
Comment (dkalinin) [inline app/store/actions.js]: Please handle server error.

**Comment by Oleh-Stefanovych** [comment]
@dkalinin, fixed, PTAL.

File: app/component/high-order-component/files-page/index.js [hunk 6]
```diff
@@ -52,9 +57,19 @@ export class FilesPage extends HighOrderComponent {
...
+          })
+          .catch((error) => this._showServerMessage(error.message));
     }).onFileDrop((files, targetFolder) => {
-      this.store.dispatch('uploadFiles', files, targetFolder);
+      console.log(files);
+      this.store.dispatch('uploadFiles', files, targetFolder)
+          .then(() => {
+            this.store.dispatch('fetchFiles', this.store.state.currentFolder)
+                .catch((error) => this._showServerMessage(error.message));
+          });
```
Comment (dkalinin) [inline app/component/high-order-component/files-page/index.js]: Please add catch to uploadFiles promise.

**Pull request merged**